File: app/src/main/java/com/example/map/data/repositoryimpl/FavoriteDocumentRepositoryImpl.java
package com.example.map.data.repositoryimpl;

import com.example.map.data.local.dao.FavoriteDocumentDao;
import com.example.map.data.local.model.DocumentEntity;
import com.example.map.domain.repository.FavoriteDocumentRepository;

import java.util.List;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Flowable;

public class FavoriteDocumentRepositoryImpl implements FavoriteDocumentRepository {
    private final FavoriteDocumentDao dao;

    public FavoriteDocumentRepositoryImpl(FavoriteDocumentDao dao) {
        this.dao = dao;
    }

    @Override
    public Flowable<List<DocumentEntity>> getAll() {
        return dao.getAll();
    }

    @Override
    public Flowable<DocumentEntity> get(String id) {
        return dao.get(id);
    }

    @Override
    public Completable insert(DocumentEntity documentEntity) {
        return dao.insert(documentEntity);
    }

    @Override
    public Completable delete(DocumentEntity documentEntity) {
        return dao.delete(documentEntity);
    }
}

File: app/src/main/java/com/example/map/data/repositoryimpl/FavoriteDocumentRepositoryImpl.kt
package com.example.map.data.repositoryimpl

import com.example.map.data.local.dao.FavoriteDocumentDao
import com.example.map.data.local.model.DocumentEntity
import com.example.map.domain.repository.FavoriteDocumentRepository
import dagger.hilt.android.scopes.ViewModelScoped
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Flowable
import javax.inject.Inject

@ViewModelScoped
class FavoriteDocumentRepositoryImpl @Inject constructor(
    private val dao: FavoriteDocumentDao
) : FavoriteDocumentRepository {
    override fun getAll(): Flowable<List<DocumentEntity>> {
        return dao.getAll()
    }

    override fun get(id: String): Flowable<DocumentEntity> {
        return dao.get(id)
    }

    override fun insert(documentEntity: DocumentEntity): Completable {
        return dao.insert(documentEntity)
    }

    override fun delete(documentEntity: DocumentEntity): Completable {
        return dao.delete(documentEntity)
    }
}
